test/support/nats_server_control.js

```diff
@@ -102,9 +103,24 @@ function start_server(port, opt_flags, done) {
 
 exports.start_server = start_server;
 
-function stop_server(server) {
-    if (server !== undefined) {
+function wait_stop(server, done) {
+    if(server.killed) {
+        if(done) {
+            done();
+        }
+    } else {
+        setTimeout(function () {
+            wait_stop(server, done);
+        });
+    }
+}
+
+function stop_server(server, done) {
+    if (server) {
```

Comment:
If `stop_server` is always called with a server, you should not test for `server` and let it fail hard so we can fix the test.

Reply:
We looked at the test suite and there are cases with clustering that we end up calling stop with a null reference.